org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/JavaSearchScope.java

```diff
@@ -239,19 +241,24 @@ public void add(IJavaElement element) throws JavaModelException {
 			String projectPath = root.getJavaProject().getPath().toString();
 			if (rootResource != null && rootResource.isAccessible()) {
 				String relativePath = Util.relativePath(rootResource.getFullPath(), containerPath.segmentCount());
-				add(projectPath, relativePath, containerPathToString, false/*not a package*/, null);
+				add(projectPath, relativePath, containerPathToString, null, false/*not a package*/, null);
 			} else {
-				add(projectPath, "", containerPathToString, false/*not a package*/, null); //$NON-NLS-1$
+				add(projectPath, "", containerPathToString, null, false/*not a package*/, null); //$NON-NLS-1$
 			}
 			break;
 		case IJavaElement.PACKAGE_FRAGMENT:
 			root = (PackageFragmentRoot)element.getParent();
 			projectPath = root.getJavaProject().getPath().toString();
 			if (root.isArchive()) {
 				String relativePath = Util.concatWith(((PackageFragment) element).names, '/');
+				IModuleDescription moduleDescription = root.getModuleDescription();
+				String moduleName = null;
+				if (moduleDescription != null) {
+					moduleName = moduleDescription.getElementName();
+				}
 				containerPath = root.getPath();
 				containerPathToString = containerPath.getDevice() == null ? containerPath.toString() : containerPath.toOSString();
-				add(projectPath, relativePath, containerPathToString, true/*package*/, null);
+				add(projectPath, relativePath, containerPathToString, moduleName, true/*package*/, null);
```

Comment:
I'm guessing, more calls to ```add``` will need the module name passed. If we find more bugs due to including the module name in the path arguments here. For now I think its best to fix 1 thing at a time.

Reply:
Right, I've found #401 while experimenting with this patch.